### DISCUSSION TITLE
How can I get annotations of a method parameter?

### DISCUSSION THREAD

**Discussion by Liyw979** [soot-oss/SootUp#718] [Q&A] [answered]
```
interface XXXDao {
    int sqlQuery(@Param("a") Long a, @Param("b") String b, @Param("c") Long c);
}
```

**Comment by swissiety** [comment]
You can try to access it via the Local that is stored in the IdentityStmt for the parameters; try casting the Local to JavaLocal and then there is a getter for Annotations.

**Reply by Liyw979** [reply]
`sqlQuery` is abstract and implemented by mybatis dynamiclly. So I am not able to access the IdentityStmt

**Reply by swissiety** [reply, marked as answer]
Oh I see - then we have a problem. I'll create an Issue that this case is not covered.

**Comment by JonasKlauke** [comment]
First you have to get the method:
`view.getMethod(
              JavaIdentifierFactory.getInstance()
                  .getMethodSignature(
                      view.getClass("XXXDao"),
                      "sqlQuery",
                      "int",
                      Arrays.asList("Long", "String", "Long"))`
then you get the method body of the method
`methodFromBefore.getBody()`
then you can get the parameters
`JavaLocal parameterLocal = (JavaLocal) body.getParameterLocal(0);
parameterLocal.getAnnotations()
`

you can find an example in the AnnotationUsageTest in method testAnnotationOnLocal()

Edit: i have not tested the code above. there might be some mistakes

**Reply by turn1tup** [reply]
It really work?
"There is no corresponding body if the method is not concrete i.e." 
I met this error.

**Comment by Liyw979** [comment]
I think there should be a class `Parameter` to model this, and allow user to do like 

- `method.getParameter(index).getAnnotations`
- `method.getParameter(index).getName`
